lib/unified_search_builder.rb

```diff
@@ -22,11 +23,14 @@ def payload
     hash_without_blank_values(
       from: search_params.start,
       size: search_params.count,
-      fields: search_params.return_fields,
+      # `title` and `description` always needed to potentially populate virtual
+      # fields. If not explicitly requested they will not be sent to the user.
+      fields: search_params.return_fields + %w[title description],
```

Comment:
I feel that these extra fields ought only to be requested if the corresponding virtual fields are requested.